src/GafferScene/SceneAlgo.cpp

```diff
@@ -614,6 +628,31 @@ void addCopyAttributesPredecessors( const CopyAttributes *copyAttributes, const
 	}
 }
 
+void addCopyOptionsPredecessors( const CopyOptions *copyOptions, const SceneAlgo::History::Predecessors &source, SceneAlgo::OptionHistory *destination )
+{
+	const ScenePlug *sourceScene = copyOptions->inPlug();
+	if( StringAlgo::matchMultiple( destination->optionName, copyOptions->optionsPlug()->getValue() ) )
+	{
+		ConstCompoundObjectPtr sourceOptions;
+		if( copyOptions->sourcePlug()->exists() )
+		{
+			sourceOptions = copyOptions->sourcePlug()->globalsPlug()->getValue();
+		}
+		if( sourceOptions && sourceOptions->members().count( g_optionPrefix + destination->optionName.string() ) )
```

Comment:
Once we've done the above, we can omit `sourceOptions &&` since we'll always have a valid pointer (if the `source` input doesn't have a connection, we'll still have got the default value for the globals which is an empty map).